tools/cldr-code/src/main/java/org/unicode/cldr/tool/FindFlipFlops.java

```diff
@@ -0,0 +1,204 @@
+package org.unicode.cldr.tool;
+
+import com.google.common.base.Joiner;
+import com.ibm.icu.util.ICUException;
+import com.ibm.icu.util.VersionInfo;
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import org.unicode.cldr.test.DisplayAndInputProcessor;
+import org.unicode.cldr.tool.Option.Options;
+import org.unicode.cldr.util.CLDRFile;
+import org.unicode.cldr.util.Factory;
+import org.unicode.cldr.util.LanguageTagParser;
+import org.unicode.cldr.util.SimpleFactory;
+
+public class FindFlipFlops {
+
+    private static int flipFlopCount = 0;
+
+    private static CldrVersion[] VERSIONS;
+
+    private static Factory[] factories;
+
+    private static final boolean USE_RESOLVED = false;
+
+    static final Options myOptions = new Options()
+        .add(
+            "file",
+            ".*",
+            ".*",
+            "Filter the information based on file name, using a regex argument. The '.xml' is removed from the file before filtering"
+        )
+        .add("oldest", "\\d+(\\.\\d+)?(\\.\\d+)?", "1.1", "Oldest version to go back to, eg 36.1");
+
+    public static void main(String[] args) throws IOException {
+        myOptions.parse(args, true);
+        confirmVersionArchiveIsPresent();
+        VERSIONS = makeVersionArray();
+        factories = setUpFactories();
+
+        testFindFlipFlop();
+
+        // load and process English
+        System.out.println("en");
+        findFlops("en");
+
+        // Load and process all the locales
+        LanguageTagParser ltp = new LanguageTagParser();
+        for (String fileName : factories[0].getAvailable()) {
+            if (fileName.equals("en")) {
+                continue;
+            }
+            if (!ltp.set(fileName).getRegion().isEmpty()) {
+                continue; // skip region locales
+            }
+            System.out.println(fileName);
+            findFlops(fileName);
+        }
+        System.out.println("Total " + flipFlopCount + " flip-flops");
+    }
+
+    private static void confirmVersionArchiveIsPresent() {
+        try {
```

Comment:
some of these methods are based on code in GenerateBirth.java -- if this pr is accepted, some code should go in a shared utility class, in a subsquent pr